src/ripple/app/tx/impl/CashCheck.cpp

```diff
@@ -365,10 +367,15 @@ CashCheck::doApply ()
             }
 
             // Make sure that deliverMin was satisfied.
-            if (optDeliverMin && result.actualAmountOut < *optDeliverMin)
+            if (optDeliverMin)
             {
-                JLOG(ctx_.journal.warn()) << "flow did not produce DeliverMin.";
-                return tecPATH_PARTIAL;
+                if (result.actualAmountOut < *optDeliverMin)
+                {
+                    JLOG(ctx_.journal.warn()) << "flow did not produce DeliverMin.";
+                    return tecPATH_PARTIAL;
+                }
+                // Set the delivered_amount metadata.
+                ctx_.deliver (result.actualAmountOut);
```

Comment:
This looks possibly wrong to me. Are you setting a nonzero delivered amount even in the case where you return `tecPATH_PARTIAL`?

Reply:
This looks correct to me. If we return `tecPATH_PARTIAL` on line 375 we never get to this line. Or did you mean something else?

Reply:
Oh, no, my bad. I just misread the "return" statement!